File: MonoGame/Other CLR Languages/MyMonoGameInVBNet/MyMonoGameInVBNet/Game1.vb
Imports Microsoft.Xna.Framework
Imports Microsoft.Xna.Framework.Graphics
Imports Microsoft.Xna.Framework.Input

Public Class Game1
    Inherits Game

    Private graphics As GraphicsDeviceManager
    Private spriteBatch As SpriteBatch

    Public Sub New()
        graphics = New GraphicsDeviceManager(Me)
        graphics.PreferredBackBufferWidth = 800
        graphics.PreferredBackBufferHeight = 600
        Window.Title = "MonoGame with Visual Basic .Net"
        Content.RootDirectory = "Content"
    End Sub

    Protected Overrides Sub Initialize()
        MyBase.Initialize()
    End Sub

    Protected Overrides Sub LoadContent()
        MyBase.LoadContent()
    End Sub

    Protected Overrides Sub Update(gameTime As GameTime)
        MyBase.Update(gameTime)

        If Keyboard.GetState().IsKeyDown(Keys.Escape) Or GamePad.GetState(PlayerIndex.One).IsButtonDown(Buttons.Back) Then
            Me.Exit()
        End If
    End Sub

    Protected Overrides Sub Draw(gameTime As GameTime)
        GraphicsDevice.Clear(Color.CornflowerBlue)
        MyBase.Draw(gameTime)
    End Sub
End Class

File: MonoGame/Other CLR Languages/MyMonoGameInCpp/MyMonoGameInCpp/Game1.cpp
#include "Game1.h"

using namespace Microsoft::Xna::Framework;
using namespace Microsoft::Xna::Framework::Graphics;
using namespace Microsoft::Xna::Framework::Input;

Game1::Game1(void) : Game()
{
	graphics = gcnew GraphicsDeviceManager(this);
	graphics->PreferredBackBufferWidth = 800;
	graphics->PreferredBackBufferHeight = 600;
	Window->Title = "MonoGame C++/Cli";
	Content->RootDirectory = "Content";
}

Game1::~Game1(void)
{
}

void Game1::Initialize()
{
	Game::Initialize();
}

void Game1::LoadContent()
{
	Game::LoadContent();
	spriteBatch = gcnew SpriteBatch(GraphicsDevice);
}

void Game1::UnloadContent()
{
	Game::UnloadContent();
}

void Game1::Update(GameTime ^gameTime)
{
	Game::Update(gameTime);
}

void Game1::Draw(GameTime ^gameTime)
{
	GraphicsDevice->Clear(Color::CornflowerBlue);
	Game::Draw(gameTime);
}


